lib/galaxy/managers/interactivetool.py

```diff
@@ -18,125 +26,97 @@
-            finally:
-                conn.close()
+        columns = ("token", "host", "port", "info")
+
+        with self._engine.connect() as conn:
+            query = select(*gxitproxy.c[columns]).where(
+                gxitproxy.c["key"] == key,
+                gxitproxy.c["key_type"] == key_type,
+            )
+            result = conn.execute(query).fetchone()
+
+        return dict(key=key, key_type=key_type, **dict(zip(columns, result))) if result else None
 
     def save(self, key, key_type, token, host, port, info=None):
         """
-        Writeout a key, key_type, token, value store that is can be used for coordinating
-        with external resources.
+        Write out a key, key_type, token, value store that is can be used for coordinating with external resources.
         """
         assert key, ValueError("A non-zero length key is required.")
         assert key_type, ValueError("A non-zero length key_type is required.")
         assert token, ValueError("A non-zero length token is required.")
-        with FileLock(self.sqlite_filename):
-            conn = sqlite3.connect(self.sqlite_filename)
-            try:
-                c = conn.cursor()
-                try:
-                    # Create table
-                    c.execute(
-                        f"""CREATE TABLE {DATABASE_TABLE_NAME}
-                                 (key text,
-                                  key_type text,
-                                  token text,
-                                  host text,
-                                  port integer,
-                                  info text,
-                                  PRIMARY KEY (key, key_type)
-                                  )"""
-                    )
-                except Exception:
-                    pass
-                delete = f"""DELETE FROM {DATABASE_TABLE_NAME} WHERE key=? and key_type=?"""
-                c.execute(
-                    delete,
-                    (
-                        key,
-                        key_type,
-                    ),
-                )
-                insert = f"""INSERT INTO {DATABASE_TABLE_NAME}
-                            (key, key_type, token, host, port, info)
-                            VALUES (?, ?, ?, ?, ?, ?)"""
-                c.execute(
-                    insert,
-                    (
-                        key,
-                        key_type,
-                        token,
-                        host,
-                        port,
-                        info,
-                    ),
-                )
-                conn.commit()
-            finally:
-                conn.close()
+        with self._engine.connect() as conn:
+            # create database table if not exists
+            gxitproxy.create(conn, checkfirst=True)
+
+            # delete existing data with same key
+            query_delete = delete(gxitproxy).where(
+                gxitproxy.c["key"] == key,
```

Comment:
```suggestion
                gxitproxy.c.key == key,
```
You can access the columns directly in all such cases. The key lookup would be more useful if you needed a subset of the table column collection, like `gxitproxy.c["key", "key_type"]`

Reply:
I assume you use this style everywhere so I changed the lookup type in 83050167 for consistency